### PR TITLE
Fixed PS-5353 (Wrong binlog entry for BLOB on a blackhole intermediary master) (8.0)

### PR DESCRIPTION
**Pull request by percona-ysorokin**

### CODE REVIEW

**Comment by percona-ysorokin** [comment]
Jenkins build link:
https://ps80.cd.percona.com/job/percona-server-8.0-param/84/

File: storage/blackhole/ha_blackhole.cc [hunk 1]
```diff
@@ -128,9 +129,49 @@ int ha_blackhole::rnd_next(uchar *) {
...
+      overwrites BLOB data in BI.
+
+      To prevent this we do the same trick as for virtual generated columns
+      in 5.7 - keeping old BLOB value inside 'old_value' field in
+      'Field_blob' class by calling 'keep_old_value()' for all BLOB fields
+      currently marked for update in 'table->write_set'..
+    */
+
+    if (table_share->blob_fields != 0)
+      for (Field **field_ptr = table->field; *field_ptr != NULL; ++field_ptr) {
```
Comment (laurynas-biveinis) [inline storage/blackhole/ha_blackhole.cc]: C++11'ize

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]

File: storage/blackhole/ha_blackhole.cc (hunk 1 above)
Comment (percona-ysorokin) [reply storage/blackhole/ha_blackhole.cc]: fixed

**Comment by percona-ysorokin** [comment]
Updated Jenkins build link
https://ps80.cd.percona.com/job/percona-server-8.0-param/91/ (rpl.bug93917 rpl.bug93917 --repeat=2)

**Review by laurynas-biveinis — APPROVED** [review]

**Pull request merged**